app/functoria_app.ml

```diff
@@ -573,10 +572,13 @@ module Make (P: S) = struct
     let root = Cmd.realpath (Filename.dirname file) in
     let file = root / Filename.basename file in
     set_config_file file;
-    compile_and_dynlink file >>= fun () ->
-    registered () >>= fun t ->
-    Log.set_section (Config.name t);
-    Ok t
+    try
+      compile_and_dynlink file >>= fun () ->
+      registered () >>= fun t ->
+      Log.set_section (Config.name t);
+      Ok t
+    with Config'.No_base_key_map ->
+      Error "Access to base key map is unavailable at this point"      
```

Comment:
Maybe we should specify which "point" we are talking about ? Something like "Access to base key map is not available before configuration. For dynamically configured devices, please use the configurable class." ?